client/my-sites/customer-home/controller.jsx

```diff
@@ -38,24 +37,10 @@ export async function maybeRedirect( context, next ) {
 	}
 
 	const siteId = getSelectedSiteId( state );
-	const maybeStalelaunchpadScreenOption = getSiteOptions( state, siteId )?.launchpad_screen;
+	const { launchpad_screen, site_intent } = await fetchLaunchpad( slug );
 
-	// We need the latest site info to determine if a user should be redirected to Launchpad.
-	// As a result, we can't use locally cached data, which might think that Launchpad is
-	// still enabled when, in reality, the final tasks have been completed and everything is
-	// disabled. Because of this, we refetch site information and limit traffic by scoping down
-	// requests to launchpad enabled sites.
-	// See https://cylonp2.wordpress.com/2022/09/19/question-about-infinite-redirect/#comment-1731
-	if (
-		( maybeStalelaunchpadScreenOption && maybeStalelaunchpadScreenOption === 'full' ) ||
-		getQueryArgs()?.showLaunchpad === 'true'
-	) {
-		await context.store.dispatch( requestSite( siteId ) );
-	}
-
-	const refetchedOptions = getSiteOptions( context.store.getState(), siteId );
 	const shouldRedirectToLaunchpad =
-		refetchedOptions?.launchpad_screen === 'full' &&
+		launchpad_screen === 'full' &&
 		// Temporary hack/band-aid to resolve a stale issue with atomic that the requestSite
```

Comment:
Wondering if we need the 'temporary band-aid for atomic' part of this given the new endpoint?

Reply:
We can remove the `shouldRedirectToLaunchpad` logic since the `const { launchpad_screen, site_intent } = await fetchLaunchpad( slug );` should always have the latest data.